### ISSUE TITLE
View not updated after ionViewWillEnter or ionViewDidEnter in Ionic2 RC.0

### ISSUE THREAD

**Issue by peterpeterparker** [driftyco/ionic#8352]
View is not updated after ionViewWillEnter or ionViewDidEnter model change in Ionic2 RC.0

ts example:

    @Component({
        templateUrl: 'test.html',
    })
    export class TestPage {
        test:string='false';
    
       ionViewWillEnter() {
           this.test = 'true';
       }
     }

html example:

      <ion-header></ion-header>    
      <ion-content>
           {{test}}
      </ion-content>

=> displayed will always be

      false

Am I the only one facing that problem? Is that a bug? If no, what should be done instead?

**Comment by peterpeterparker** [comment]
I tried the same example in the starter app and it worked, but unfortunately didn't find any way to solve that in my app.

Furthermore, I also noticed I've got some refresh problem too with other component, like the following:

ts:
     import {Component, Input} from '@angular/core';
    
    @Component({
        templateUrl: 'image-loader.html',
        selector: 'image-loader'
    })
    export class ImageLoader {
        @Input() src;
    
        loaded:boolean = false;
    
        imageLoaded() {
            this.loaded = true;
        }
    
    }

html:
    <ion-spinner *ngIf="!loaded"></ion-spinner>

    <img [hidden]="!loaded" (load)="imageLoaded()" [src]="src"/>

    
I added an alert to image loaded and I effectively see that "this.loaded = true;" is set but still nothing happens in the browser.

Seems that both problem are related to me, so is it like there is a configuration to tell Ionic2/Angular2 how to survey and refresh according the model changes?

**Comment by manucorporat** [comment]
Looking into this issue. can you test if it does update if wrapping your changes around:

```
this.zone.run(() => {
        this.loaded = true;
});
```

you should inject NgZone. Obviously this is only a workaround. Let me know it that fixed the issue.

**Comment by peterpeterparker** [comment]
First thx a log for looking this issue @manucorporat !!!

I tried following. Unfortunately didn't worked or helped. Did I miss something?

     import {Component, Input, NgZone} from '@angular/core';
     
     @Component({
         templateUrl: 'image-loader.html',
         selector: 'image-loader'
     })
     export class ImageLoader {
         @Input() src;
     
         loaded:boolean = false;
     
         constructor(public zone:NgZone) {
     
         }
     
         imageLoaded() {
             this.zone.run(() => {
                 this.loaded = true;
                 console.log(this.loaded);
             });
         }
     
     }

**Comment by peterpeterparker** [comment]
I also just did a test on my login page and there the update work.
So I guess that I do something wrong in the meantime which break the zone update or something like that...gonna continue to dig

**Comment by peterpeterparker** [comment]
Don't know what but what is wrong is something with the component I include in other component.

In pages where I do not use other component, like that ImageLoader, I don't have any problem with update and binding of the model. As soon as I include one single component  of my own, then these update/zone problem occurs.

**Comment by peterpeterparker** [comment]
Well then found nothing more. Tried to reproduce the problem with the starter app, can't :(

Seems I really have got problems with my custom components, as for event firing as for updating the gui.

The other task I opened is most probably linked with that one...

https://github.com/driftyco/ionic/issues/8345

Any idea is most welcomed in advance

**Comment by peterpeterparker** [comment]
F*CK found the problem I think.

So basically on my login page I've got an "automatic login" which trigger a login with a cordova plugin. This crash an error in the browser, it always did, but wasn't a problem because I was working with enableProdMode() which isn't the case anymore.

Since this error crash all other occurring afterwards aren't displayed in the console. I removed this "auto login" and could discover where my app was crashing for real aka where my problem was.

Actually in two different custom components, an error "Expression has changed after it was checked" is throw. Not fixed the error but commented the error part and everything is back to normal.

How to loose 6 hours of your life...at least I found where the problem is.

Sorry for bothering everybody who read this post, specially @manucorporat
